_posts/2022-07-06-how-to-use-airflow-to-schedule-trino-jobs.md

```diff
@@ -0,0 +1,376 @@
+$ mkdir dags plugins logs
+```
+
+Step 3) Download the Airflow docker compose yaml file.
+
+```
+$ curl -LfO 'https://airflow.apache.org/docs/apache-airflow/stable/docker-compose.yaml'
+```
+
+Step 4) Create an `.env` configuration file: 
+
+```
+$ echo -e "AIRFLOW_UID=$(id -u)" > .env
+$ echo "AIRFLOW_GID=0" >> .env 
+```
+
+Step 5) Start the Airflow containers
+
+```
+$ docker-compose up -d
+```
+
+# Installing the TrinoHook
+
+The best practice, if running Airflow in docker, is to install the TrinoHook in
+all the docker containers using the `apache/airflow:x.x.x` image.
+
+```
+$ docker ps 
+CONTAINER ID   IMAGE                  PORTS                              NAMES
+cffdfaeb757e   apache/airflow:2.3.0   0.0.0.0:8080->8080/tcp             airflow_airflow-webserver_1
+b0e72f479a66   apache/airflow:2.3.0   8080/tcp                           airflow_airflow-worker_1
+4cdb11b3e5e3   apache/airflow:2.3.0   8080/tcp                           airflow_airflow-triggerer_1
+41d3c3107ddb   apache/airflow:2.3.0   0.0.0.0:5555->5555/tcp, 8080/tcp   airflow_flower_1
+229a11e9cdd3   apache/airflow:2.3.0   8080/tcp                           airflow_airflow-scheduler_1
+68160240857d   postgres:13            5432/tcp                           airflow_postgres_1
+a96b98da85df   redis:latest           6379/tcp                           airflow_redis_1
+```
+
+To install the TrinoHook you run `pip install apache-airflow-providers-trino` in
+the first five containers.  Run the following command replacing the container id of
+each of the containers in your deployment.
+
+```
+$ docker exec -it <container_id> pip install apache-airflow-providers-trino
+```
+
+Once you have done that you need to restart all five containers:
+
+```
+$ docker container restart <container_id_1> ... <container_id_5>
+```
+
+# Creating a Trino connection
+
+After you have installed the TrinoHook and restarted Airflow you can create a
+connection to your Trino cluster through the Airflow web UI.  If you just
+install Airflow then go to `http://localhost:8080` on your browser and login.
+The default credentials unless changed are `airflow` for username and password.
+
+Go to Admin > Connections.
+
+<p align="center">
+   <img align="center" width="75%" src="/assets/blog/trino-airflow-blog/airflow-connections.png"/>
+</p>
+
+Click on the blue button to **Add a new record**.
+
+<p align="center">
+   <img align="center" width="75%" src="/assets/blog/trino-airflow-blog/airflow-new-connection.png"/>
+</p>
+
+Select **Trino** from the Connection Type dropdown and provide the following information:
+
+<table>
+  <tr>
+   <td>Connection Id</td>
+   <td>Whatever you want to call your connection.</td>
+  </tr>
+  <tr>
+   <td>
+    Host
+   </td>
+   <td>The hostname or host ip of your trino cluster, e.g., <code>localhost</code>, <code>10.10.10.1</code>, or <code>www.mytrino.com</code></td>
+  </tr>
+  <tr>
+   <td>Schema</td>
+   <td>A schema in your Trino cluster, not important which.</td>
```

Comment:
Maybe simply `A schema in your Trino cluster` or `A default schema in your Trino cluster`?